opstack/src/consensus.rs

```diff
@@ -268,140 +270,65 @@ fn payload_to_block(value: ExecutionPayload) -> Result<Block<Transaction>> {
-                        nonce: inner_tx.nonce(),
-                        block_hash: Some(value.block_hash),
-                        block_number: Some(value.block_number),
-                        transaction_index: Some(i as u64),
-                        to: inner_tx.to().to().cloned(),
-                        value: inner_tx.value(),
-                        gas_price: inner_tx.gas_price(),
-                        gas: inner_tx.gas_limit(),
-                        input: inner_tx.input().to_vec().into(),
-                        chain_id: inner_tx.chain_id(),
-                        transaction_type,
-                        from: inner.recover_signer()?,
-                        signature: Some(Signature {
-                            r: inner.signature().r(),
-                            s: inner.signature().s(),
-                            v: U256::from(inner.signature().v().to_u64()),
-                            y_parity: Some(Parity(inner.signature().v().to_u64() == 1)),
-                        }),
-                        access_list: Some(inner_tx.tx().access_list.clone()),
-                        max_fee_per_gas: Some(inner_tx.tx().max_fee_per_gas),
-                        max_priority_fee_per_gas: Some(inner_tx.tx().max_priority_fee_per_gas),
-                        max_fee_per_blob_gas: Some(inner_tx.tx().max_fee_per_blob_gas),
-                        blob_versioned_hashes: Some(inner_tx.tx().blob_versioned_hashes.clone()),
-                        ..Default::default()
+                        inner: inner_tx,
+                        deposit_nonce: None,
+                        deposit_receipt_version: None,
                     }
                 }
                 OpTxEnvelope::Deposit(inner) => {
-                    let hash =
-                        keccak256([&[0x7Eu8], alloy::rlp::encode(&inner).as_slice()].concat());
-
-                    let tx = Transaction {
-                        hash,
-                        nonce: inner.nonce(),
-                        block_hash: Some(value.block_hash),
-                        block_number: Some(value.block_number),
-                        transaction_index: Some(i as u64),
-                        to: inner.to().to().cloned(),
-                        value: inner.value(),
-                        gas_price: inner.gas_price(),
-                        gas: inner.gas_limit(),
-                        input: inner.input().to_vec().into(),
-                        chain_id: inner.chain_id(),
-                        transaction_type,
-                        ..Default::default()
-                    };
-
-                    tx
+                    inner_tx.from = inner.inner().from;
+                    let deposit_nonce = Some(inner.inner().nonce());
+
+                    Transaction {
+                        inner: inner_tx,
+                        deposit_nonce,
+                        deposit_receipt_version: None,
```

Comment:
I've left it as `None` for now. Or do we want to implement a check for Canyon fork in scope of this PR?